features/lib/step_definitions/scenarios_and_steps.rb

```diff
@@ -30,26 +30,26 @@ def snake_case(name)
 Given('a scenario {string} that passes') do |name|
   create_feature(name) do
     create_scenario(name) do
-      '  Given it passes'
+      "  Given it passes in #{name}"
```

Comment:
Do we need these changes here? Just trying to think how we can diet down the complexity?

Reply:
To make my new scenario [`retry_failing_tests.feature:99`](https://github.com/cucumber/cucumber-ruby/blob/d6c5c1e257a2881883c800c3e49ef212f10a7f06/features/docs/cli/retry_failing_tests.feature#L99) pass I only would need to fix `scenarios_and_steps.rb:43` and not all.

The problem was that the steps could only be used once per scenario, my fix makes it possible to use such a step several times in one scenario.

I therefore could only fix `scenarios_and_steps.rb:43` and not all but I thought it's better if all steps behave and are implemented the same.

Is that ok or do you want me to only fix the step that caused the problem?

Reply:
@luke-hill I tried to connect to cucumberbdd.slack.com but it seems I need an account there. Can you send me invite to jorg.jenni@jennius.co.uk?

Reply:
As an update and from what others have said it's likely we'll just implement this here as we've done for now. I'll take a look and review this in the coming week.

You should be able to sign up on your own to the slack channel